commit 4e47c3ca1fd460a3000a58cd2c37ee0a91f77247
Author: Keno Fischer <keno@juliacomputing.com>
Date:   Thu Jul 4 19:33:56 2019 -0400

    [cmake] Don't set install rules for tblgen if building utils is disabled
    
    Summary:
    This is a follow up to D64032. Afterwards if building utils is disabled
    and cross compilation is attempted, CMake will complain that adding
    `install()` directives to targets with EXCLUDE_FROM_ALL set is "undefined".
    Indeed, it appears depending on the CMake version and the selected
    Generator, the install rule will error because the underlying target isn't
    built. Fix that by not adding the install rule if building utils is not
    requested. Note that this doesn't prevent building tblgen as a
    dependency in not cross-build, even if building tools is disabled.
    
    Reviewers: smeenai
    
    Subscribers: mgorny, llvm-commits
    
    Tags: #llvm
    
    Differential Revision: https://reviews.llvm.org/D64225

diff --git a/cmake/modules/TableGen.cmake b/cmake/modules/TableGen.cmake
index 095b316d627..75985ba905e 100644
--- a/cmake/modules/TableGen.cmake
+++ b/cmake/modules/TableGen.cmake
@@ -173,7 +173,7 @@ macro(add_tablegen target project)
   add_llvm_executable(${target} DISABLE_LLVM_LINK_LLVM_DYLIB ${ARGN})
   set(LLVM_LINK_COMPONENTS ${${target}_OLD_LLVM_LINK_COMPONENTS})

-  if (${project} STREQUAL LLVM AND NOT LLVM_INSTALL_TOOLCHAIN_ONLY)
+  if (${project} STREQUAL LLVM AND NOT LLVM_INSTALL_TOOLCHAIN_ONLY AND LLVM_BUILD_UTILS)
     if(${target} IN_LIST LLVM_DISTRIBUTION_COMPONENTS OR
         NOT LLVM_DISTRIBUTION_COMPONENTS)
       set(export_to_llvmexports EXPORT LLVMExports)
